source/application/AMPIC887DriverServer/GCS2Commands/AMGCS2GetLowSoftLimitsCommand.cpp

```diff
@@ -2,50 +2,34 @@
 #include "../AMGCS2Support.h"
 #include "util/AMCArrayHandler.h"
 #include "PI_GCS2_DLL.h"
-AMGCS2GetLowSoftLimitsCommand::AMGCS2GetLowSoftLimitsCommand(const QList<AMGCS2::Axis>& axes)
+#include "../AMPIC887Controller.h"
+AMGCS2GetLowSoftLimitsCommand::AMGCS2GetLowSoftLimitsCommand(const AMPIC887AxisCollection& axesToQuery )
 {
-	axesToQuery_ = axes;
+	axesToQuery_ = axesToQuery;
 }
 
-QHash<AMGCS2::Axis, double> AMGCS2GetLowSoftLimitsCommand::axesLowSoftLimits() const
+AMPIC887AxisMap<double> AMGCS2GetLowSoftLimitsCommand::axesLowSoftLimits() const
 {
 	return axesLowSoftLimits_;
 }
 
-QString AMGCS2GetLowSoftLimitsCommand::outputString() const
+bool AMGCS2GetLowSoftLimitsCommand::validateArguments()
```

Comment:
It looks like this validate arguments method is exactly the same in all subclasses.  There is probably a way to simplify that code so that it's not repeated over and over again.